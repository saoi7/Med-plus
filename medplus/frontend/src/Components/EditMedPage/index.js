import AddMedPage from "../AddMedPage";
import { withFirebase } from "../Firebase";
import ModifyMedForm from "../ModifyMedForm";
import NavBar from "../NavBar";
import { withRouter } from 'react-router-dom';

function EditMedPageBase(props) {
    let times_to_take_arr = Object.entries(props.location.state.times_to_take).map(entry => {
        let [time_to_take, quantity] = entry;
        return {
            time_to_take,
            quantity
        };
    });
    const initial_state = {
        ...props.location.state,
        edit_page_flag: true,
        page_title: "Edit medication",
        submit_button_text: "Apply changes",
        times_to_take: times_to_take_arr,
    };
    
    return (
        <div className="background-with-logo-image add-med-layout">
            <div className="title font-large">
                { initial_state.page_title }
            </div>
            <ModifyMedForm firebase={props.firebase} initialState={initial_state} />
            <NavBar />
        </div>
    );
}

const EditMedPage = withRouter(withFirebase(EditMedPageBase));
export default EditMedPage;

export { EditMedPageBase };
