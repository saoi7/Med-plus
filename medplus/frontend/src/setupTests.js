// jest-dom adds custom jest matchers for asserting on DOM nodes.
// allows you to do things like:
// expect(element).toHaveTextContent(/react/i)
// learn more: https://github.com/testing-library/jest-dom
import '@testing-library/jest-dom';
import { unmountComponentAtNode } from "react-dom";

let test_dom_container = null;
beforeEach(() => {
  // setup a DOM element as a render target
  test_dom_container = document.createElement("div");
  document.body.appendChild(test_dom_container);
});

afterEach(() => {
  // cleanup on exiting
  unmountComponentAtNode(test_dom_container);
  test_dom_container.remove();
  test_dom_container = null;
});

export { test_dom_container };
