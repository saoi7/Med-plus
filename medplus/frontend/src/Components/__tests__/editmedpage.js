import { MOCK_DB_MED_ENTRIES } from './mock_db_med_entries';
import {test_dom_container} from '../../setupTests';
import { render, unmountComponentAtNode } from "react-dom";
import { act } from "react-dom/test-utils";
import renderer from 'react-test-renderer';
import { getByTestId, fireEvent } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import { MOCK_FIREBASE_PROP } from './mock_firebase_props';
import flushPromises from 'flush-promises';
import { EditMedPageBase } from '../EditMedPage';

test("snapshot test of EditMedPage", () => {
    const MOCK_LOCATION_STATE_PROP = {
        "edit_page_flag": true,
        "end_date": "2021-12-14",
        "med_name": "abc",
        "page_title": "Edit medication",
        "start_date": "2021-11-01",
        "submit_button_text": "Apply changes",
        "times_to_take": {
            "04:00": "1",
            "05:00": "999"
        }
    };
    const COMPONENT= renderer.create(
        <MemoryRouter>
            <EditMedPageBase location={{state: MOCK_LOCATION_STATE_PROP }} firebase={MOCK_FIREBASE_PROP}/>
        </MemoryRouter>
    );
    let tree = COMPONENT.toJSON();
    expect(tree).toMatchSnapshot();
});
