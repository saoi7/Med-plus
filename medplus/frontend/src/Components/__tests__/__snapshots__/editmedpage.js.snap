// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test of EditMedPage 1`] = `
<div
  className="background-with-logo-image add-med-layout"
>
  <div
    className="title font-large"
  >
    Edit medication
  </div>
  <form
    className="add-med-form"
    onSubmit={[Function]}
  >
    <div
      className="add-med-input font-small undefined"
    >
      <label
        className="left-padding"
      >
        Name: 
      </label>
      <input
        className="hidden-text-box left-padding font-small"
        name="med_name"
        onChange={[Function]}
        required={true}
        type="text"
        value="abc"
      />
    </div>
    <div
      className="add-med-input font-small undefined"
    >
      <label
        className="left-padding"
      >
        Start Date: 
      </label>
      <input
        className="hidden-text-box left-padding font-small"
        name="start_date"
        onChange={[Function]}
        required={true}
        type="date"
        value="2021-11-01"
      />
    </div>
    <div
      className="add-med-input font-small undefined"
    >
      <label
        className="left-padding"
      >
        End Date: 
      </label>
      <input
        className="hidden-text-box left-padding font-small"
        name="end_date"
        onChange={[Function]}
        required={true}
        type="date"
        value="2021-12-14"
      />
    </div>
    <div
      className="med-list-time-to-take-quantity-group"
    >
      <div
        className="add-med-input font-small undefined"
      >
        <label
          className="left-padding"
        >
          Time To Take: 
        </label>
        <input
          className="hidden-text-box left-padding font-small"
          name="time_to_take"
          onChange={[Function]}
          required={true}
          type="time"
          value="04:00"
        />
      </div>
      <div
        className="add-med-input font-small quantity-input"
      >
        <label
          className="left-padding"
        >
          Quantity: 
        </label>
        <input
          className="hidden-text-box left-padding font-small"
          name="quantity"
          onChange={[Function]}
          required={true}
          type="number"
          value="1"
        />
      </div>
      <svg
        className="time-to-take-button"
        fill="currentColor"
        height="1em"
        onClick={[Function]}
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        version="1.1"
        viewBox="0 0 16 16"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 6.5v3c0 0.276 0.224 0.5 0.5 0.5h15c0.276 0 0.5-0.224 0.5-0.5v-3c0-0.276-0.224-0.5-0.5-0.5h-15c-0.276 0-0.5 0.224-0.5 0.5z"
        />
      </svg>
    </div>
    <div
      className="med-list-time-to-take-quantity-group"
    >
      <div
        className="add-med-input font-small undefined"
      >
        <label
          className="left-padding"
        >
          Time To Take: 
        </label>
        <input
          className="hidden-text-box left-padding font-small"
          name="time_to_take"
          onChange={[Function]}
          required={true}
          type="time"
          value="05:00"
        />
      </div>
      <div
        className="add-med-input font-small quantity-input"
      >
        <label
          className="left-padding"
        >
          Quantity: 
        </label>
        <input
          className="hidden-text-box left-padding font-small"
          name="quantity"
          onChange={[Function]}
          required={true}
          type="number"
          value="999"
        />
      </div>
      <svg
        className="time-to-take-button"
        data-testid="add-time-to-take-button"
        fill="currentColor"
        height="1em"
        onClick={[Function]}
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        version="1.1"
        viewBox="0 0 16 16"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M15.5 6h-5.5v-5.5c0-0.276-0.224-0.5-0.5-0.5h-3c-0.276 0-0.5 0.224-0.5 0.5v5.5h-5.5c-0.276 0-0.5 0.224-0.5 0.5v3c0 0.276 0.224 0.5 0.5 0.5h5.5v5.5c0 0.276 0.224 0.5 0.5 0.5h3c0.276 0 0.5-0.224 0.5-0.5v-5.5h5.5c0.276 0 0.5-0.224 0.5-0.5v-3c0-0.276-0.224-0.5-0.5-0.5z"
        />
      </svg>
    </div>
    <div
      className="add-med-input font-small"
    >
      <input
        className="hidden-submit-button font-small"
        type="submit"
        value="Apply changes"
      />
    </div>
  </form>
  <nav
    className="nav-bar"
  >
    <a
      className="font-medium"
      href="/home"
      onClick={[Function]}
    >
      Reminders
    </a>
    <a
      className="font-medium"
      href="/home"
      onClick={[Function]}
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 20 20"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M10.707 2.293a1 1 0 00-1.414 0l-7 7a1 1 0 001.414 1.414L4 10.414V17a1 1 0 001 1h2a1 1 0 001-1v-2a1 1 0 011-1h2a1 1 0 011 1v2a1 1 0 001 1h2a1 1 0 001-1v-6.586l.293.293a1 1 0 001.414-1.414l-7-7z"
        />
      </svg>
    </a>
    <a
      className="font-medium"
      href="/account"
      onClick={[Function]}
    >
      Profile
    </a>
  </nav>
</div>
`;
